cell_classification/predict.py

```diff
@@ -0,0 +1,236 @@
+import toml
+import argparse
+import pickle
+import os
+from sklearn.metrics import roc_curve, auc, confusion_matrix
+from model_builder import ModelBuilder
+import numpy as np
+from tqdm import tqdm
+from plot_utils import plot_average_roc, plot_metrics_against_threshold
+import toml
+from copy import deepcopy
+from joblib import Parallel, delayed
+import pandas as pd
+
+
+def load_model_and_val_data(params):
+    """Load model and validation data from params dict
+    Args:
+        params (dict):
+            dictionary containing model and validation data paths
+    Returns:
+        model (ModelBuilder):
+            trained model
+        val_data (tf.data.Dataset):
+            validation dataset
+    """
+    params["eval"] = True
+    model = ModelBuilder(params)
+    model.prep_data()
+    model.load_model(params["model_path"])
+    val_dset = iter(model.validation_dataset)
+    return model, val_dset
+
+
+def predict(model, val_dset, params):
+    """Predict labels for validation data
+    Args:
+        model (ModelBuilder):
+            trained model
+        val_dset (tf.data.Dataset):
+            validation data
+        params (dict):
+            dictionary containing model and validation data paths
+    Returns:
+        predictions (list):
+            list of dictionaries containing the full example + prediction
+        params (dict):
+            dictionary containing model hyperparams and validation data paths
+    """
+    pred_list = []
```

Comment:
Is this eventually going to be a problem keeping all of the validation set in memory?

Reply:
When the dataset goes 100x it will be a problem. I'll serialize results in this function individually, probably via .hdf instead of .pkl then it can be loaded with fiji.

Reply:
Got it, sounds good